src/lib/storage/storage_manager.cpp

```diff
@@ -5,42 +5,48 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(!has_table(name), "Tried to create table with existing name.");
+  _tables[name] = std::move(table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(has_table(name), "Tried to drop a non existing table.");
+  _tables.erase(name);
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  Assert(has_table(name), "Tried to get a non existing table.");
+  return _tables.at(name);
 }
 
 bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _tables.contains(name);
 }
 
 std::vector<std::string> StorageManager::table_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto table_names = std::vector<std::string>();
```

Comment:
```suggestion
  auto table_names = std::vector<std::string>{};
```
I think this style of creating the vector is preferred for consistency reasons

Reply:
Hm, I am not sure about this one. The first one to me says "make a default vector (by calling the default constructor)" whereas the second says "make a vector from this (empty) list" (in theory, it calls the vector constructor for `initializer_list`, but shouldn't matter once optimizations are on). What do the others think?